`StargateV1ConfigurationSourceProviderTest` failing on v2 branch

```
Step #4 - "build": [INFO] Running io.stargate.metrics.jersey.dwconfig.StargateV1ConfigurationSourceProviderTest
Step #4 - "build": INFO [StargateV1ConfigurationSourceProvider](module=testapi): No value for configuration override System Property 'stargate.configurationFile.testapi'
Step #4 - "build": INFO [StargateV1ConfigurationSourceProvider](module=testapi): No configuration override file '/workspace/metrics-jersey/testapi-config.yaml' found
Step #4 - "build": INFO [StargateV1ConfigurationSourceProvider](module=testapi): No configuration overrides found, will use the default config resource 'testapi-config.yaml'
Step #4 - "build": INFO [StargateV1ConfigurationSourceProvider](module=nosuchmodule): No value for configuration override System Property 'stargate.configurationFile.nosuchmodule'
Step #4 - "build": INFO [StargateV1ConfigurationSourceProvider](module=nosuchmodule): No configuration override file '/workspace/metrics-jersey/noconfig.yaml' found
Step #4 - "build": INFO [StargateV1ConfigurationSourceProvider](module=nosuchmodule): No configuration overrides found, will use the default config resource 'noconfig.yaml'
Step #4 - "build": INFO [StargateV1ConfigurationSourceProvider](module=testapi): Found configuration override System Property 'stargate.configurationFile.testapi'; will use config file 'src/test/resources/alt-config.yaml'
Step #4 - "build": INFO [StargateV1ConfigurationSourceProvider](module=testapi): No value for configuration override System Property 'stargate.configurationFile.testapi'
Step #4 - "build": INFO [StargateV1ConfigurationSourceProvider](module=testapi): No configuration override file '/workspace/metrics-jersey/testapi-config.yaml' found
Step #4 - "build": INFO [StargateV1ConfigurationSourceProvider](module=testapi): No configuration overrides found, will use the default config resource 'testapi-config.yaml'
Step #4 - "build": [ERROR] Tests run: 4, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.139 s <<< FAILURE! - in io.stargate.metrics.jersey.dwconfig.StargateV1ConfigurationSourceProviderTest
Step #4 - "build": [ERROR] testNamingConventionBased  Time elapsed: 0.016 s  <<< FAILURE!
Step #4 - "build": org.opentest4j.AssertionFailedError: 
Step #4 - "build": 
Step #4 - "build": expected: "config: custom"
Step #4 - "build":  but was: "config: default"
Step #4 - "build": 	at io.stargate.metrics.jersey.dwconfig.StargateV1ConfigurationSourceProviderTest.testNamingConventionBased(StargateV1ConfigurationSourceProviderTest.java:57)
Step #4 - "build": 
Step #4 - "build": [INFO] Running io.stargate.metrics.jersey.listener.CounterRequestEventListenerTest
Step #4 - "build": [INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.65 s - in io.stargate.metrics.jersey.listener.CounterRequestEventListenerTest
Step #4 - "build": [INFO] Running io.stargate.metrics.jersey.listener.CounterApplicationEventListenerTest
Step #4 - "build": [INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.766 s - in io.stargate.metrics.jersey.listener.CounterApplicationEventListenerTest
Step #4 - "build": [INFO] Running io.stargate.metrics.jersey.config.SystemPropsMetricsListenerConfigTest
Step #4 - "build": [INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.107 s - in io.stargate.metrics.jersey.config.SystemPropsMetricsListenerConfigTest
Step #4 - "build": [INFO] Running io.stargate.metrics.jersey.MetricsBinderTest
Step #4 - "build": [INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.794 s - in io.stargate.metrics.jersey.MetricsBinderTest
Step #4 - "build": [INFO] 
Step #4 - "build": [INFO] Results:
Step #4 - "build": [INFO] 
Step #4 - "build": [ERROR] Failures: 
Step #4 - "build": [ERROR]   StargateV1ConfigurationSourceProviderTest.testNamingConventionBased:57 
Step #4 - "build": expected: "config: custom"
Step #4 - "build":  but was: "config: default"
Step #4 - "build": [INFO] 
Step #4 - "build": [ERROR] Tests run: 50, Failures: 1, Errors: 0, Skipped: 0
```

--- a/metrics-jersey/src/test/java/io/stargate/metrics/jersey/dwconfig/StargateV1ConfigurationSourceProviderTest.java
+++ b/metrics-jersey/src/test/java/io/stargate/metrics/jersey/dwconfig/StargateV1ConfigurationSourceProviderTest.java
@@ -5,6 +5,8 @@ import static org.assertj.core.api.Assertions.assertThat;
 import java.io.File;
 import java.io.InputStream;
 import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.nio.file.Paths;
 import org.apache.commons.io.FileUtils;
 import org.apache.commons.io.IOUtils;
 import org.junit.jupiter.api.Test;
@@ -48,12 +50,8 @@ public class StargateV1ConfigurationSourceProviderTest {
     // ensuring it is not seen during other tests
     final String originalCWD = System.getProperty("user.dir");
     try {
-      File f = new File(originalCWD);
-      f = new File(f, "src");
-      f = new File(f, "test");
-      f = new File(f, "resources");
-      f = new File(f, "cwd");
-      String value = f.toString();
+      Path path = Paths.get(originalCWD, "src", "test", "resources", "cwd");
+      String value = path.toAbsolutePath().toString();
       System.out.printf("Setting user.dir to point to %s%n", value);
       System.setProperty("user.dir", value);
       assertThat(readConfig()).isEqualTo("config: custom");

--- a/metrics-jersey/src/test/java/io/stargate/metrics/jersey/dwconfig/StargateV1ConfigurationSourceProviderTest.java
+++ b/metrics-jersey/src/test/java/io/stargate/metrics/jersey/dwconfig/StargateV1ConfigurationSourceProviderTest.java
@@ -53,7 +53,9 @@ public class StargateV1ConfigurationSourceProviderTest {
       f = new File(f, "test");
       f = new File(f, "resources");
       f = new File(f, "cwd");
-      System.setProperty("user.dir", f.toString());
+      String value = f.toString();
+      System.out.printf("Setting user.dir to point to %s%n", value);
+      System.setProperty("user.dir", value);
       assertThat(readConfig()).isEqualTo("config: custom");
     } finally {
       System.setProperty("user.dir", originalCWD);

--- a/metrics-jersey/src/test/java/io/stargate/metrics/jersey/dwconfig/StargateV1ConfigurationSourceProviderTest.java
+++ b/metrics-jersey/src/test/java/io/stargate/metrics/jersey/dwconfig/StargateV1ConfigurationSourceProviderTest.java
@@ -5,6 +5,7 @@ import static org.assertj.core.api.Assertions.assertThat;
 import java.io.File;
 import java.io.InputStream;
 import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import org.apache.commons.io.FileUtils;
@@ -46,17 +47,16 @@ public class StargateV1ConfigurationSourceProviderTest {
   // Test to verify that inclusion of an alternate configuration based on naming convention
   @Test
   public void testNamingConventionBased() throws Exception {
-    // Bit unclean as we need be able to co-locate alternative config file to our CWD,
-    // ensuring it is not seen during other tests
-    final String originalCWD = System.getProperty("user.dir");
+    // simply copy src/test/resources/cwd/testapi-config.yaml
+    // to the working directory
+    Path configFile = Paths.get("src", "test", "resources", "cwd", "testapi-config.yaml");
+    Path targetFile = Paths.get("testapi-config.yaml");
+
     try {
-      Path path = Paths.get(originalCWD, "src", "test", "resources", "cwd");
-      String value = path.toAbsolutePath().toString();
-      System.out.printf("Setting user.dir to point to %s%n", value);
-      System.setProperty("user.dir", value);
+      Files.copy(configFile, targetFile);
       assertThat(readConfig()).isEqualTo("config: custom");
     } finally {
-      System.setProperty("user.dir", originalCWD);
+      Files.deleteIfExists(targetFile);
     }
   }
 

